e2e/testcases/oci_sync_test.go

```diff
@@ -35,56 +36,74 @@ import (
-	//│   └── role.yaml
-	//├── kustomization.yaml
-	//├── README.md
-	//├── tenant-a
-	//│   └── kustomization.yaml
-	//├── tenant-b
-	//│   └── kustomization.yaml
-	//├── tenant-c
-	//│   └── kustomization.yaml
-	//└── tenant-d
-	//    └── kustomization.yaml
-	// Root kustomization.yaml content:
-	// #kustomization.yaml
-	// resources:
-	// - tenant-a
-	// - tenant-b
-	// - tenant-c
-	// - tenant-d
-	// Base kustomization.yaml content:
-	// resources:
-	// - namespace.yaml
-	// - rolebinding.yaml
-	// - role.yaml
-	// - networkpolicy.yaml
-	// commonLabels:
-	//  test-case: hydration
-
-	// publicGCRImage pulls the public OCI image by the default `latest` tag
-	publicGCRImage = "gcr.io/stolos-dev/config-sync-ci/kustomize-components"
-	// publicARImage pulls the public OCI image by the default `latest` tag
-	publicARImage = "us-docker.pkg.dev/stolos-dev/config-sync-ci-public/kustomize-components"
-	// privateGCRImage pulls the private OCI image by digest
-	privateGCRImage = "us.gcr.io/stolos-dev/config-sync-ci/kustomize-components@sha256:e5b67f70a0cb3e7afb539521739baece7b8918282ccd6ef6be7caa061df95a3b"
-	// privateARImage pulls the private OCI image by tag
-	privateARImage    = "us-docker.pkg.dev/stolos-dev/config-sync-ci-private/kustomize-components:v1"
-	imageDigest       = "e5b67f70a0cb3e7afb539521739baece7b8918282ccd6ef6be7caa061df95a3b"
-	gsaARReaderEmail  = "e2e-test-ar-reader@stolos-dev.iam.gserviceaccount.com"
-	gsaGCRReaderEmail = "e2e-test-gcr-reader@stolos-dev.iam.gserviceaccount.com"
+	imageDigest = "e5b67f70a0cb3e7afb539521739baece7b8918282ccd6ef6be7caa061df95a3b"
 )
 
+// All the following images include a kustomization.yaml file in the root directory,
+// which includes resources for tenant-a, tenant-b, tenant-c, tenant-d.
+// Each tenant includes a NetworkPolicy, a Role and a RoleBinding.
+// Image structure:
+// .
+//├── base
+//│   ├── kustomization.yaml
+//│   ├── namespace.yaml
+//│   ├── networkpolicy.yaml
+//│   ├── rolebinding.yaml
+//│   └── role.yaml
+//├── kustomization.yaml
+//├── README.md
+//├── tenant-a
+//│   └── kustomization.yaml
+//├── tenant-b
+//│   └── kustomization.yaml
+//├── tenant-c
+//│   └── kustomization.yaml
+//└── tenant-d
+//    └── kustomization.yaml
+// Root kustomization.yaml content:
+// #kustomization.yaml
+// resources:
+// - tenant-a
+// - tenant-b
+// - tenant-c
+// - tenant-d
+// Base kustomization.yaml content:
+// resources:
+// - namespace.yaml
+// - rolebinding.yaml
+// - role.yaml
+// - networkpolicy.yaml
+// commonLabels:
+//  test-case: hydration
+
+// publicGCRImage pulls the public OCI image by the default `latest` tag
+func publicGCRImage() string {
+	return fmt.Sprintf("gcr.io/%s/config-sync-ci/kustomize-components", os.Getenv("GCP_PROJECT"))
+}
+
+// publicARImage pulls the public OCI image by the default `latest` tag
+func publicARImage() string {
+	return fmt.Sprintf("us-docker.pkg.dev/%s/config-sync-ci-public/kustomize-components", os.Getenv("GCP_PROJECT"))
```

Comment:
/hold

I noticed we have some yaml that references to images here as well ([example](https://github.com/GoogleContainerTools/kpt-config-sync/blob/main/e2e/testdata/reconciler-manager/reposync-oci.yaml#L23)). Do we need to support providing the project at runtime for that yaml?

One a similar note, we also have test images like [git-server](https://github.com/GoogleContainerTools/kpt-config-sync/blob/main/Makefile.build#L319) and [buildenv](https://github.com/GoogleContainerTools/kpt-config-sync/blob/main/Makefile#L81) which are only published in `stolos-dev`. Should we make these generic?

Reply:
The [example](https://github.com/GoogleContainerTools/kpt-config-sync/blob/main/e2e/testdata/reconciler-manager/reposync-oci.yaml#L23) is only used for manually installing and configuring Config Sync, so it won't affect the e2e test.

For  `git-server` and `buildenv`, I explicitly leave them unchanged. Those are prebuilt public images. We can continue using them until we decide to tear down the `stolos-dev` project.

Reply:
Ack, SGTM

Reply:
https://github.com/GoogleContainerTools/kpt-config-sync/blob/main/e2e/testdata/reconciler-manager/reposync-oci.yaml#L23 is indeed used in the test. I updated this PR to remove the YAML file. The OCI spec was configured in the test.